### PR TITLE
Bump Matter Server to 2.0.1 and set Vendor ID

### PR DESCRIPTION
**Pull request by agners**
Not to be merged until HA 2023.2 beta 0 builds done

### CODE REVIEW

File: matter_server/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 3.0.0
```
Comment (balloob) [inline matter_server/CHANGELOG.md:3]: This says 3, message and pr description say 2

Comment (balloob) [reply matter_server/CHANGELOG.md:3]: Ah addon 3, matter server 2

**Review by balloob — APPROVED** [review]

File: matter_server/rootfs/etc/s6-overlay/s6-rc.d/matter-server/run [hunk 2]
```diff
@@ -26,4 +26,4 @@ fi
 cd /root
 
 exec matter-server --storage-path "/data" --port "${server_port}" \
-                   --log-level "${log_level}"
+                   --log-level "${log_level}" --vendorid 4939
```
Comment (MartinHjelmare) [inline matter_server/rootfs/etc/s6-overlay/s6-rc.d/matter-server/run:29]: Looks correct in the diagnostics of my test device:

```
"0/62/1": {
          "node_id": 3,
          "endpoint": 0,
          "cluster_id": 62,
          "cluster_type": "chip.clusters.Objects.OperationalCredentials",
          "cluster_name": "OperationalCredentials",
          "attribute_id": 1,
          "attribute_type": "chip.clusters.Objects.OperationalCredentials.Attributes.Fabrics",
          "attribute_name": "Fabrics",
          "value": [
            {
              "rootPublicKey": "BGzv50odoWbboMwqnL2say8QhLGya6eA4KvT3wDL2RPzjBD5pqU7PVsUflToxrpgjO9q+HlhocQjWH/pHaO0KZQ=",
              "vendorId": 4939,
              "fabricId": 1,
              "nodeId": 3,
              "label": "",
              "fabricIndex": 1
            }
          ]
        },
```

**Review by MartinHjelmare — APPROVED** [review]
I've tested building this version of the add-on locally and commissioned a test device. Working good so far. :+1:

File: matter_server/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
...
+
+- Bump Matter Server to 2.0.1
+- Use Nabu Casa Vendor ID by default (requires recommissioning of devices)
```
Comment (MartinHjelmare) [inline matter_server/CHANGELOG.md:6]: Should we be a bit more explicit?

```suggestion
- Use Nabu Casa Vendor ID by default (requires recommissioning of devices)

### Breaking

- All commissioned devices need to be recommissioned.
```

Comment (agners) [reply matter_server/CHANGELOG.md:6]: Yeah a bit more explicit seems sensible. I guess then we can drop it from the brackets above.

```suggestion
- Use Nabu Casa Vendor ID by default

### Breaking

- All commissioned devices need to be recommissioned.
```

**Pull request merged**